packages/editor/src/Sx/Typography.tsx

```diff
@@ -36,12 +48,13 @@ export const SxTypography = ({
       <div
         sx={{
           display: 'grid',
-          gridGap: 2,
+          gap: 2,
           gridTemplateColumns: 'repeat(3, 1fr)',
         }}>
         <Field
           name={prefixName('fontSize')}
           label="Font Size"
+          // FIXME: Field type is comming from external package @types/theme-ui__components, not sure why value prop does not exist in type definition.
```

Comment:
It seems the `Field` doesn't default to `InputProps` properly.

Reply:
I was able to reproduce the problem with my own type (basically without `BoxOwnProps ` and `FieldOwnProps `), but couldn't figure out how to fix the issue. Is there a way to debug types easily?

I left a comment with some more info here: https://github.com/system-ui/theme-ui/pull/900#discussion_r422516796

Reply:
Okay, I just changed Field typings locally to
```tsx
export function Field<T extends React.ElementType = React.ComponentType<InputProps>>(props: FieldProps<T>): JSX.Element;
```
and to
```tsx
export function Field<T extends React.ElementType = "input">(props: FieldProps<T>): JSX.Element;
```

both work.

You can also check it when you by the generic parameter to `Field`

```tsx
      <Field<"input">
        type="number"
      />
```

```tsx
      <Field<typeof Input>
        type="number"
      />
```

Reply:
The only thing it did is skipping the iteration over `keyof JSX.IntrinsicElements` in `ElementType`.

```tsx
    type ElementType<P = any> =
        {
            [K in keyof JSX.IntrinsicElements]: P extends JSX.IntrinsicElements[K] ? K : never
        }[keyof JSX.IntrinsicElements] |
        ComponentType<P>;
```

Reply:
Okay, so there are two things to do.
Firstly, we shouldn't use `@types/theme-ui__components`.
That's our bad twin from alternative dimension and he's out to get us.
Mea culpa :D

Things we have to do.
- [ ] Move the types from DT to this repo.
       Strong deja vu -- I think actually did it once. Is it in one of the open PRs?
- [ ] Do the change I mentioned in previous comment.
- [ ] Fix my sins and PR the change I mentioned in previous comment in DT for other people
  - [ ] Reflect that in .d.ts "test" in DT.

> Is there a way to debug types easily?

Well, they usually aren't that hard to debug.

In our case -- Unit tests.
- https://github.com/cartant/ts-snippet
- https://github.com/SamVerschueren/tsd
- https://github.com/ikatyang/dts-jest

Reply:
Cool, thanks for your help! 🙌 I think I even tried `<Field<'input'>>` but didn't wait long enough for TypeScript to recheck. 😅

> Strong deja vu -- I think actually did it once. Is it in one of the open PRs?

Yes, I found it in https://github.com/system-ui/theme-ui/pull/705/files#diff-c9edcada2823b2f3423667cdb4f5bf44R209. Would you like to change it in #705 or should I import the types into this one?

Reply:
>  Do the change I mentioned in previous comment.

I fixed the Field type issues in 196bc704.

Reply:
I suppose #705 won't be merged quickly tbh.
Dunno what approach will we take with gatsby plugins.

Reply:
I didn't add the component types into my branch because I didn't need to change any of the types to get it to work. Let me know if I should still add them within this PR.